tensorbay/opendataset/PlantSeedlings/loader.py

```diff
@@ -0,0 +1,213 @@
+    "Maize": 11,
+    "Scentless Mayweed": 12,
+    "Shepherd’s Purse": 13,
+    "Sugar beet": 14,
+}
+
+
+def PlantSeedlings(path: str) -> Dataset:
+    """Dataloader of PlantSeedlings dataset.
+
+    .. PlantSeedlings: https://vision.eng.au.dk/plant-seedlings-dataset/
+
+    The file structure should be like::
+
+        <path>
+            Black-grass/
+                1.png
+                2.png
+                ...
+            Charlock/
+            Cleavers/
+            Common Chickweed/
+            Common wheat/
+            Fat Hen/
+            Loose Silky-bent/
+            Maize/
+            Scentless Mayweed/
+            Shepherd’s Purse/
+            Small-flowered Cranesbill/
+            Sugar beet/
+            Segmented/
+                Black-grass/
+                 1.png
+                 2.png
+                 ...
+                Charlock/
+                Cleavers/
+                Common Chickweed/
+                Common wheat/
+                Fat Hen/
+                Loose Silky-bent/
+                Maize/
+                Scentless Mayweed/
+                Shepherd’s Purse/
+                Small-flowered Cranesbill/
+                Sugar beet/
+
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+    dataset = Dataset("PlantSeedlings")
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+    for subset in os.listdir(root_path):
+        if subset in ("Segmented", "Fullimages", "single_channel_segmented"):
+            continue
+        dataset.add_segment(_get_segment(root_path, subset))
+    return dataset
+
+
+def PlantSeedlings_Fullimages(path: str) -> Dataset:
+    """Dataloader of PlantSeedlings_Fullimages dataset.
+
+    .. PlantSeedlings: https://vision.eng.au.dk/plant-seedlings-dataset/
+
+    The file structure should be like::
+
+        <path>
+            Fullimages/
+                1/
+                    IMG_0573_12-03.png
+                    ...
+                2/
+                    ...
+                ...
+                SpeciesIndex.pdf
+                SpeciesIndex.xlsx
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+    dataset = Dataset("PlantSeedlings_Fullimages")
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog_Fullimages.json"))
+    dataset.add_segment(_get_fullimages_segment(root_path))
+    return dataset
+
+
+def _get_fullimages_segment(root_path: str) -> Segment:
+    segment = Segment("Fullimages")
+    files_path = os.path.join(root_path, "Fullimages")
+    category_map = _get_category_map()
+    for file_path in glob(os.path.join(files_path, "[0-9]*")):
```

Comment:
can we add extension to the glob pattern like `[0-9].png`?

Reply:
[0-9]* is file folders, not the png file.